nservicebus/testing/cross-version-testing.md

```diff
@@ -0,0 +1,24 @@
+---
+title: Cross-version compatibility testing with NServiceBus
+summary: Develop service layers and long-running processes using test-driven development.
+reviewed: 2016-06-06
+related:
+ - samples/unit-testing
+---
+
+In order to ensure that NServiceBus upgrades are safe and that endpoints in various versions can communicate with each other, it is necessary to extensively test multiple aspects of cross-version compatibility. As much as possible, those tests are automated. Some of them are available publicly in the [Particular/EndToEnd](https://github.com/Particular/EndToEnd/) repository.
+
+Those tests might be useful for NServiceBus users in order to verify that the framework works correctly in an unusual scenario, for example when using an unusually complex data structure, in order to understand in detail how specific feature works, or to reproduce an issue they experienced. The provided code can be run on a local machine, the pre-requisites are listed in the [Readme](https://github.com/Particular/EndToEnd/blob/master/README.md) file in the repository.
```

Comment:
well i think u need to phrase this in terms of what they want to do and what we want to achieve

one scenario is people wanting to evolve saga data/messages moving forward. how would u write tests for this? and repeat those test over multiple iterations. while there is much code in out endtoend we could use to do that , leaving that to the user to work out how to do that is a big ask

another scenario is someone building a persister/transport/serializer. in that case there is a much higher amount of code that could be reused from endtoend

the 3rd point we want to get across is that "we do backwards compat test and we take it seriously". this is more pitched at the enterprise architect level and is almost marketing speak.